<launch>
    <!-- Launch file for kuangye hesai16 LiDAR -->
    <arg name="rviz" default="true" />
    <arg name="bagfile" default="" />
    <arg name="save_dir" default="$(find fast_lio)/Log" />
    <arg name="state_filename" default="scan_states.txt" />
    <arg name="msg_start_time" default="0" />
    <arg name="msg_end_time" default="0" />
    <arg name="stationary_start" default="true" />

    <arg name="configyaml" default="$(find fast_lio)/config/hesai32_xiangyin.yaml" />
    <rosparam command="load" file="$(arg configyaml)" />

    <param name="odom_mode" type="int" value="0" />
    <param name="bagfile" type="string" value="$(arg bagfile)" />
    <param name="state_filename" type="string" value="$(arg state_filename)" />
    <param name="msg_start_time" type="string" value="$(arg msg_start_time)" />
    <param name="msg_end_time" type="string" value="$(arg msg_end_time)" />
    <param name="stationary_start" type="bool" value="$(arg stationary_start)" />
    <param name="save_dir" type="string" value="$(arg save_dir)" />
    <node pkg="fast_lio" type="fastlio_mapping" name="laserMapping" output="screen" required="true" /> 

    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="fastlio_rviz" args="-d $(find fast_lio)/rviz_cfg/loam_livox.rviz" />
    </group>

</launch>
